blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClient.scala

```diff
@@ -112,27 +112,31 @@ object BlazeClient {
                     }
                 }
 
-              Deferred[F, Unit].flatMap { gate =>
-                val responseHeaderTimeoutF: F[TimeoutException] =
-                  F.delay {
-                      val stage =
-                        new ResponseHeaderTimeoutStage[ByteBuffer](
-                          responseHeaderTimeout,
-                          scheduler,
-                          ec)
-                      next.connection.spliceBefore(stage)
-                      stage
-                    }
-                    .bracket(stage =>
-                      F.asyncF[TimeoutException] { cb =>
-                        F.delay(stage.init(cb)) >> gate.complete(())
-                    })(stage => F.delay(stage.removeStage()))
+              responseHeaderTimeout match {
```

Comment:
scalafmt made this look bigger than it is.  In short, just return the response if there's nothing finite to race against.